File: architecture-lab/src/main/java/org/hyeonqz/architecturelab/eda/domainevent/order/domain/Order.java
package org.hyeonqz.architecturelab.eda.domainevent.order.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "eda_order")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String customerId;

    @Column(nullable = false)
    private Long amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status;

    public static Order create(String customerId, Long amount) {
        Order order = new Order();
        order.customerId = customerId;
        order.amount = amount;
        order.status = OrderStatus.PLACED;
        return order;
    }
}

File: kotlin-lab/src/main/kotlin/org/hyeonqz/kotlinlab/kotlinlab/osiv/domain/Order.kt
package org.hyeonqz.kotlinlab.kotlinlab.osiv.domain

import jakarta.persistence.*

@Entity
@Table(name = "orders")
class Order(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    val amount: Long,

    val status: String = "PENDING",

    // Lazy 연관 — OSIV 테스트의 핵심 포인트
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "merchant_id")
    val merchant: Merchant,

    @OneToMany(mappedBy = "order", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
    val items: MutableList<OrderItem> = mutableListOf()
)